pkg/kubectl/sorting_printer.go

```diff
@@ -103,7 +103,9 @@ func SortObjects(decoder runtime.Decoder, objs []runtime.Object, fieldInput stri
 		switch u := item.(type) {
 		case *runtime.Unknown:
 			var err error
-			if objs[ix], _, err = decoder.Decode(u.Raw, nil, nil); err != nil {
+			// decode runtime.Unknown to runtime.Unstructured for sorting.
+			// we don't actually want the internal versions of known types.
+			if objs[ix], _, err = decoder.Decode(u.Raw, nil, &runtime.Unstructured{}); err != nil {
```

Comment:
I don't suppose there is anything that tests this?  From david's generic get work?